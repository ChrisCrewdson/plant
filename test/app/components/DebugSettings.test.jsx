
const React = require('react');
const renderer = require('react-test-renderer');
const MuiThemeProvider = require('material-ui/styles/MuiThemeProvider').default;
const getMuiTheme = require('material-ui/styles/getMuiTheme').default;
const lightBaseTheme = require('material-ui/styles/baseThemes/lightBaseTheme').default;
const {
  MemoryRouter,
} = require('react-router-dom');
const { Provider } = require('react-redux');
const DebugSettings = require('../../../app/components/DebugSettings');
const store = require('../../../app/store');
const App = require('../../../app/components/App');

const muiTheme = getMuiTheme(lightBaseTheme);

describe('DebugSettings', () => {
  const storeDispatch = store.dispatch;

  beforeAll(() => {
    store.dispatch = () => {};
    global.localStorage = {
      getItem: () => '',
      setItem: () => {},
    };
  });

  afterAll(() => {
    store.dispatch = storeDispatch;
  });

  test('DebugSettings should be rendered', () => {
    const component = renderer.create(
      <MuiThemeProvider muiTheme={muiTheme}>
        <Provider store={store}>
          <App>
            <MemoryRouter>
              <DebugSettings />
            </MemoryRouter>
          </App>
        </Provider>
      </MuiThemeProvider>);
    const badgeTotals = component.toJSON();
    expect(badgeTotals).toMatchSnapshot();
  });
});
