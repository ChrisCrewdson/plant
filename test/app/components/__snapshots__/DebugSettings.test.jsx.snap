// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DebugSettings DebugSettings should be rendered 1`] = `
<div
  className="react-root"
>
  <div
    className="page"
  >
    <nav
      className="navbar navbar-default navbar-fixed-top"
    >
      <div
        className="container-fluid"
      >
        <div
          className="navbar-header"
        >
          <button
            aria-expanded="false"
            className="navbar-toggle collapsed"
            data-target="#plant-navbar-collapse"
            data-toggle="collapse"
            type="button"
          >
            <span
              className="sr-only"
            >
              Toggle navigation
            </span>
            <span
              className="icon-bar"
            />
            <span
              className="icon-bar"
            />
            <span
              className="icon-bar"
            />
          </button>
          <a
            className="navbar-brand"
            href="/"
            onClick={[Function]}
          >
            Plant
          </a>
        </div>
        <div
          className="collapse navbar-collapse"
          id="plant-navbar-collapse"
        >
          <ul
            className="nav navbar-nav navbar-right"
          >
            <li>
              <a
                href="/login"
                onClick={[Function]}
              >
                Login
              </a>
            </li>
            <li>
              <a
                href="/help"
                onClick={[Function]}
                title="help"
              >
                Help
              </a>
            </li>
          </ul>
        </div>
      </div>
    </nav>
    <div
      id="main"
    />
    <nav
      className="navbar navbar-default navbar-fixed-bottom"
    >
      <div
        className="container-fluid"
      >
        <a
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
        <a
          href="/privacy"
          onClick={[Function]}
          style={
            Object {
              "marginLeft": "10px",
            }
          }
        >
          Privacy
        </a>
        <a
          href="/terms"
          onClick={[Function]}
          style={
            Object {
              "marginLeft": "10px",
            }
          }
        >
          Terms
        </a>
      </div>
    </nav>
  </div>
</div>
`;
